rwengine/src/ai/CharacterController.cpp

```diff
@@ -252,12 +252,12 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
                     character->getCurrentState().armour = 100.f;
                 } else if (vehicle->getVehicle()->vehiclename_ == "AMBULAN") {
                     // Receive up to 25 HPs from Ambulance
-                    if (character->getCurrentState().health < 100.f) {
-                        if (character->getCurrentState().health > 75.f) {
-                            character->getCurrentState().health = 100.f;
-                        } else {
-                            character->getCurrentState().health += 25.f;
-                        }
+                    float health = character->getCurrentState().health;
+                    // todo: Replace by correct fullHealth once implemented
+                    float fullHealth = 100.f;
+                    if (health < fullHealth) {
+                        character->getCurrentState().health =
+                            (health >= 80.f) ? fullHealth : health + 20.f;
```

Comment:
* The comment does not match this anymore
* I'd suggest something like `std::min(health + 20.f, fullHealth)` but I see that it could give problems with prostitute health boosts which temporarily raise the health higher than fullHealth. So it looks like we'd still require a check `if (health < fullHealth)` to avoid lowering the health.
* You still have to check with the original game wether this also happens `if (health > 100.f)` when fullHealth is greater than 100.

Optionally create issues after merge so we can check this later